bleak/backends/bluezdbus/discovery.py

```diff
@@ -132,99 +268,69 @@ def parse_msg(message):
-        signature="a{sv}",
-        body=[filters],
-    ).asFuture(loop)
-
-    await bus.callRemote(
-        adapter_path,
-        "StartDiscovery",
-        interface="org.bluez.Adapter1",
-        destination="org.bluez",
-    ).asFuture(loop)
+async def discover_async(callback: Callable[[BLEDevice], None]=None,
+                         loop=None, **kwargs):
+    """Start discovering asynchronously nearby Bluetooth Low Energy devices.
+    The filters are applied and the callback registered with the object is
+    called every time a new device appears or the properties of an already
+    discovered device changes. This might happen frequently, since a change in
+    the RSSI value is considered a property change.
 
-    await asyncio.sleep(timeout)
+    Args:
+        callback (Callable[[bleak.BLEDevice], None]): called for each discovered device.
+        loop (asyncio.AbstractEventLoop): Optional event loop to use.
 
-    await bus.callRemote(
-        adapter_path,
-        "StopDiscovery",
-        interface="org.bluez.Adapter1",
-        destination="org.bluez",
-    ).asFuture(loop)
-
-    # Reduce output.
-    discovered_devices = []
-    for path, props in devices.items():
-        if not props:
-            logger.debug(
-                "Disregarding %s since no properties could be obtained." % path
-            )
-            continue
-        name, address, _, path = _device_info(path, props)
-        if address is None:
-            continue
-        uuids = props.get("UUIDs", [])
-        manufacturer_data = props.get("ManufacturerData", {})
-        discovered_devices.append(
-            BLEDevice(
-                address,
-                name,
-                {"path": path, "props": props},
-                uuids=uuids,
-                manufacturer_data=manufacturer_data,
-            )
-        )
+    Keyword Args:
+        device (str): Bluetooth device to use for discovery.
+        filters (dict): A dict of filters to be applied on discovery.
+
+    Returns:
+        A discovery state object that can be used to stop the discovery again.
+
+    """
+    device = kwargs.get("device", "hci0")
 
-    for rule in rules:
-        await bus.delMatch(rule).asFuture(loop)
+    # Discovery filters
+    filters = kwargs.get("filters", {})
+
+    disco = AsyncDiscovery(callback, loop, device)
+    await disco._start_discovery(filters)
 
-    bus.disconnect()
+    return disco
+
+
+async def discover(timeout=5.0, loop=None, **kwargs):
+    """Discover nearby Bluetooth Low Energy devices.
+
+    For possible values for `filter`, see the parameters to the
+    ``SetDiscoveryFilter`` method in the `BlueZ docs
+    <https://git.kernel.org/pub/scm/bluetooth/bluez.git/tree/doc/adapter-api.txt?h=5.48&id=0d1e3b9c5754022c779da129025d493a198d49cf>`_
 
-    return discovered_devices
+    The ``Transport`` parameter is always set to ``le`` by default in Bleak.
+
+    Args:
+        timeout (float): Duration to scan for.
```

Comment:
maybe consider making it clear that its only the scan?
```suggestion
       scan_timeout (float): Duration in seconds to scan for.
```

Reply:
This is the BlueZ implementation of the discover API. I would have to change this for all implementations and still break the API. I'd rather not do that ;)